src/SA/dir/entry.c

```diff
@@ -0,0 +1,103 @@
+#include <windows.h>
+#include "bofdefs.h"
+#include "base.c"
+
+void listDir(wchar_t *path) {
+
+	WIN32_FIND_DATAW fd = {0};
+	HANDLE hand = NULL;
+	LARGE_INTEGER fileSize;
+	LONGLONG totalFileSize = 0;
+	int nFiles = 0;
+	int nDirs = 0;
+	
+	// If the file ends in \ or is a drive (C:), throw a * on there
+	int a = MSVCRT$wcslen(path);
```

Comment:
This may not crash, but it isn't memory safe.  we don't control how path was allocated by BeaconExtract, so if implementation changes, or we don't just get lucky it could crash.

I recommend making either a reasonably sized stack variable (1024ish to account for unicode paths) Or doing an allocation after your wcslen + a few bytes if you need to do the strcats and freeing it at the end.  if you do the stack var, make sure to use one of the variants of the string functions that specify max size.

Reply:
I think I followed what you meant. main() now copies out the data to a separate variable to we pass in a specified max number of bytes, and there should only ever be one recursive call(line 33). When I was experimenting with a recursive dir, the recursive stack was blowing up but it should handle a depth of 1.

Reply:
yep you hit that fun item.  If you check out sc_enum you'll see how I worked around recursion in that one as well.  I think max stack size for BOF is 4K so it fills up quick if you're not careful.

If we're not talking about unicode paths the standard max filepath is defined as MAX_PATH which typically is 260.  That said, 1K for a quick dir is fine and helps acount for if someone ever did need to do a unicode path